### ISSUE TITLE
Ruby Segmentation Fault

### ISSUE THREAD

**Issue by blowmage** [google/protobuf#3788]
This morning the google-cloud-ruby project had a segfault on the Travis-CI build system.

System details:
 
* gRPC language and version: grpc 1.6.7 (universal-darwin)
* Protobuf version: google-protobuf 3.4.1.1 (universal-darwin)
* Operating System: MacOS Sierra
* Runtime version: ruby 2.1.9p490 (2016-03-30 revision 54437) [x86_64-darwin16.0]

<details><summary>segfault details</summary>

```
/Users/travis/.rvm/gems/ruby-2.1.9/gems/google-protobuf-3.4.1.1-universal-darwin/lib/google/protobuf/duration_pb.rb:6: [BUG] Segmentation fault at 0x00000000000000
ruby 2.1.9p490 (2016-03-30 revision 54437) [x86_64-darwin16.0]

-- Crash Report log information --------------------------------------------
   See Crash Report log file under the one of following:
     * ~/Library/Logs/CrashReporter
     * /Library/Logs/CrashReporter
     * ~/Library/Logs/DiagnosticReports
     * /Library/Logs/DiagnosticReports
   for more details.

-- Control frame information -----------------------------------------------
c:0041 p:---- s:0137 e:000136 CFUNC  :build
c:0040 p:0025 s:0134 e:000133 TOP    /Users/travis/.rvm/gems/ruby-2.1.9/gems/google-protobuf-3.4.1.1-universal-darwin/lib/google/protobuf/duration_pb.rb:6 [FINISH]
c:0039 p:---- s:0132 e:000131 CFUNC  :require
c:0038 p:0015 s:0128 e:000127 TOP    /Users/travis/.rvm/gems/ruby-2.1.9/gems/google-protobuf-3.4.1.1-universal-darwin/lib/google/protobuf/well_known_types.rb:33 [FINISH]
c:0037 p:---- s:0126 e:000125 CFUNC  :require
c:0036 p:0039 s:0122 e:000121 TOP    /Users/travis/.rvm/gems/ruby-2.1.9/gems/google-gax-0.8.12/lib/google/gax/grpc.rb:34 [FINISH]
c:0035 p:---- s:0120 e:000119 CFUNC  :require
c:0034 p:0015 s:0116 e:000115 TOP    /Users/travis/.rvm/gems/ruby-2.1.9/gems/google-gax-0.8.12/lib/google/gax/errors.rb:32 [FINISH]
c:0033 p:---- s:0114 e:000113 CFUNC  :require
c:0032 p:0015 s:0110 e:000109 TOP    /Users/travis/.rvm/gems/ruby-2.1.9/gems/google-gax-0.8.12/lib/google/gax/api_callable.rb:32 [FINISH]
c:0031 p:---- s:0108 e:000107 CFUNC  :require
c:0030 p:0007 s:0104 e:000103 TOP    /Users/travis/.rvm/gems/ruby-2.1.9/gems/google-gax-0.8.12/lib/google/gax.rb:30 [FINISH]
c:0029 p:---- s:0102 e:000101 CFUNC  :require
c:0028 p:0023 s:0098 e:000097 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-video_intelligence/test/google/cloud/video_intelligence/ [FINISH]
c:0027 p:---- s:0096 e:000095 CFUNC  :require_relative
c:0026 p:0009 s:0092 e:000091 BLOCK  /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-video_intelligence/Rakefile:25 [FINISH]
c:0025 p:---- s:0089 e:000088 CFUNC  :each
c:0024 p:0030 s:0086 e:000085 BLOCK  /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-video_intelligence/Rakefile:25 [FINISH]
c:0023 p:---- s:0084 e:000083 CFUNC  :call
c:0022 p:0028 s:0079 e:000078 BLOCK  /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/task.rb:248 [FINISH]
c:0021 p:---- s:0076 e:000075 CFUNC  :each
c:0020 p:0113 s:0073 e:000072 METHOD /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/task.rb:243
c:0019 p:0075 s:0069 e:000068 BLOCK  /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/task.rb:187
c:0018 p:0014 s:0067 e:000066 METHOD /Users/travis/.rvm/rubies/ruby-2.1.9/lib/ruby/2.1.0/monitor.rb:211
c:0017 p:0025 s:0064 e:000063 METHOD /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/task.rb:180
c:0016 p:0036 s:0057 e:000056 METHOD /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/task.rb:173
c:0015 p:0033 s:0052 e:000051 METHOD /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:152
c:0014 p:0009 s:0045 e:000044 BLOCK  /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:108 [FINISH]
c:0013 p:---- s:0042 e:000041 CFUNC  :each
c:0012 p:0039 s:0039 e:000038 BLOCK  /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:108
c:0011 p:0025 s:0037 e:000036 METHOD /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:117
c:0010 p:0007 s:0033 e:000032 METHOD /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:102
c:0009 p:0019 s:0030 e:000029 BLOCK  /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:80
c:0008 p:0006 s:0028 e:000027 METHOD /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:178
c:0007 p:0007 s:0024 e:000023 METHOD /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:77
c:0006 p:0021 s:0021 e:000020 TOP    /Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/exe/rake:27 [FINISH]
c:0005 p:---- s:0019 e:000018 CFUNC  :load
c:0004 p:0140 s:0015 E:000fa8 EVAL   /Users/travis/.rvm/gems/ruby-2.1.9/bin/rake:23 [FINISH]
c:0003 p:---- s:0011 e:000010 CFUNC  :eval
c:0002 p:0118 s:0005 E:0022c0 EVAL   /Users/travis/.rvm/gems/ruby-2.1.9/bin/ruby_executable_hooks:15 [FINISH]
c:0001 p:0000 s:0002 E:001768 TOP    [FINISH]

/Users/travis/.rvm/gems/ruby-2.1.9/bin/ruby_executable_hooks:15:in `<main>'
/Users/travis/.rvm/gems/ruby-2.1.9/bin/ruby_executable_hooks:15:in `eval'
/Users/travis/.rvm/gems/ruby-2.1.9/bin/rake:23:in `<main>'
/Users/travis/.rvm/gems/ruby-2.1.9/bin/rake:23:in `load'
/Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/exe/rake:27:in `<top (required)>'
/Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:77:in `run'
/Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:178:in `standard_exception_handling'
/Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/application.rb:80:in `block in run'
/Users/travis/.rvm/gems/ruby-2.1.9/gems/rake-11.3.0/lib/rake/applica/Users/travis/.travis/job_stages: line 166: shell_session_update: command not found
```

</details>

**Comment by acozzette** [comment]
@TeBoring Is the same issue that should be fixed by #3708?

**Comment by TeBoring** [comment]
From the place of the error, ```duration_pb.rb:6```, it seems like the same error for #3708.
